lib/std/crypto/aes.zig

```diff
@@ -138,9 +146,48 @@ fn AES(comptime keysize: usize) type {
         pub fn decrypt(ctx: Self, dst: []u8, src: []const u8) void {
             decryptBlock(ctx.dec[0..], dst, src);
         }
+        pub fn ctr(ctx: Self, dst: []u8, src: []const u8, iv: [16]u8) void {
+            std.debug.assert(dst.len >= src.len);
+
+            var keystream: [16]u8 = undefined;
+            var ctrbuf = iv;
+            var n: usize = 0;
+            while (n < src.len) {
+                ctx.encrypt(keystream[0..], ctrbuf[0..]);
+                var ctr_i = std.mem.readIntSliceBig(u128, ctrbuf[0..]);
+                std.mem.writeIntSliceBig(u128, ctrbuf[0..], ctr_i +% 1);
```

Comment:
I really love that Zig lets me treat the counter as a `u128` rather than a `[16]byte`. :) In Go you'd have to write a loop to manually perform the addition.
For interoperability, though, it seemed best for the `ctr` API to accept the `iv` as a byte array.